### PR TITLE
Add Tab component and TabbedCard

### PR DESCRIPTION
**Pull request by AaronCoplan**
This relates to #111.  Added generic Tab component which can be used to build other tabbed components.  Built default TabbedCard and put it on the Cards demo page.

tab1
tab2
tab3

### CODE REVIEW

File: src/components/TabbedCard/TabbedCard.react.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  state = {
+    selectedTitle: this.props.initialTab,
+  };
+
+  render(): React.Node {
+    const tabs = React.Children.toArray(this.props.children);
+    return (
+      <Card>
+        <Card.Header>
+          <ul className="nav nav-tabs TabbedCard_header_tabs">
```
Comment (jonthomp) [inline src/components/TabbedCard/TabbedCard.react.js]: Could this be a new `Tabs` component that is built upon our existing `Nav` components?

File: src/components/TabbedCard/TabbedCard.react.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            {tabs.map(tab => {
+              const title = tab.props.title;
+              const classes =
+                title === this.state.selectedTitle
+                  ? "nav-link active"
+                  : "nav-link";
+              return (
+                <li className="nav-item">
```
Comment (jonthomp) [inline src/components/TabbedCard/TabbedCard.react.js]: Could this use our Nav.Item component?

File: src/components/TabbedCard/TabbedCard.react.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    selectedTitle: this.props.initialTab,
+  };
+
+  render(): React.Node {
+    const tabs = React.Children.toArray(this.props.children);
+    return (
+      <Card>
+        <Card.Header>
+          <ul className="nav nav-tabs TabbedCard_header_tabs">
+            {tabs.map(tab => {
```
Comment (jonthomp) [inline src/components/TabbedCard/TabbedCard.react.js]: Is it worth moving the logic and returning of the component inside here to the new Tab component? Then being able to do something like

```jsx
{tabs.map(tab => <Tab 
  title={title}
  active={title === this.state.selectedTitle} 
  onClick={() => this.setState({ selectedTitle: title })} 
/>)}
```

Of if we want to provide a bit more flexibility:
```jsx
{tabs.map(tab => React.cloneElement(tab, {
  active: { title === this.state.selectedTitle } 
  onClick: { () => this.setState({ selectedTitle: title }) } 
 })
)}
```

File: src/components/TabbedCard/TabbedCard.react.js (hunk 2 above)
Comment (jonthomp) [inline src/components/TabbedCard/TabbedCard.react.js]: I think you need to add a `key` prop to this too?

**Comment by jonthomp** [comment]
Tabs! I think this is our last 'must have' component for any website :tada:.  I only have some queries about splitting the logic out into the Tab component and maybe creating a new Tabs component, just to help make it all a bit more reusable.

**Comment by MrJarv1s** [comment]
I'd have to agree with @jonthomp, this is great news.
I have also found that the current implementation of the tabs on the website always underlines the first tab of the list (Home).

For example when viewing _Documentation_ both the Home and Documentation tabs are underlined.

**Comment by AaronCoplan** [comment]
@jonthomp I added two new components - TabbedContainer and TabbedHeader - these abstract away much of the logic.  As you may not want to render these the same way always, the state is decoupled from these two components.  All you have to do is manage state in whatever component renders both of them, which gives you the freedom to display the tabs and the content itself in two different locations.  All you have to do is include a string variable in state and have a callback in tabbed header to update it.  

You can see TabbedCard for a good example of how I used this.  Let me know your thoughts!

**Comment by jonthomp** [comment]
@AaronCoplan great! And I really like the separated TabbedContainer/TabbedHeader and so being able to place them separately :+1: 

Would you be up for adding a Tabs component as part of this PR too? I imagine that as just TabbedCard without all the Card components?

Also a query on the Tab component. As it doesn't do anything on its own it can't really be used by anyone for putting together more custom Tab setups but perhaps something we just need to make clear in comments/docs? Or is it worth extracting the Nav.Link part of Tabbedheader?

**Comment by jonthomp** [comment]
@AaronCoplan yeah exactly, then users can easily throw some ready and working tabs in wherever they like

**Comment by AaronCoplan** [comment]
@jonthomp Here are 2 screenshots of the Tabs component - should I add some padding between the header and the tabs so it doesn't look so smushed?

tabs1
tabs2

**Comment by jonthomp** [comment]
Awesome and yeah I think a bit of padding will do the trick to finish it off nicely!

**Pull request merged**